jfix-zookeeper/src/main/kotlin/ru/fix/zookeeper/lock/ActiveLocksContainer.kt

```diff
@@ -0,0 +1,117 @@
+package ru.fix.zookeeper.lock
+
+import mu.KotlinLogging
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentMap
+import java.util.concurrent.locks.ReentrantLock
+
+class ActiveLocksContainer : AutoCloseable {
+    companion object {
+        private val logger = KotlinLogging.logger { }
+    }
+
+    /**
+     * remove, prolong and getState operations MUST BE synced under [.globalLock]
+     */
+    private val locks: ConcurrentMap<LockIdentity, LockContainer> = ConcurrentHashMap()
+
+    /**
+     * all operations with accessing to [.locks] has been synced with this object
+     * LockContainer is encapsulated inside this class so for performance issue this lock could be
+     * replaced with syncing via LockContainer
+     */
+    private val globalLock = ReentrantLock(true)
+
+    /**
+     * @return - old value in the map
+     */
+    fun putLock(
+            lockId: LockIdentity,
+            lock: PersistentExpiringDistributedLock,
+            listener: LockProlongationFailedListener
+    ): PersistentExpiringDistributedLock? {
+        val newLockContainer = LockContainer(lock, listener)
+        val oldLockContainer = locks.put(lockId, newLockContainer)
+        return oldLockContainer?.lock
+    }
+
+    fun removeLock(lockId: LockIdentity): PersistentExpiringDistributedLock? {
+        return try {
+            globalLock.lock()
+            locks.remove(lockId)?.lock
+        } finally {
+            globalLock.unlock()
+        }
+    }
+
+    operator fun contains(lockId: LockIdentity): Boolean {
+        return locks.containsKey(lockId)
+    }
+
+    @Throws(Exception::class)
+    fun getLockState(lockId: LockIdentity): PersistentExpiringDistributedLock.State? {
+        return try {
+            globalLock.lock()
+            locks[lockId]?.lock?.state
+        } finally {
+            globalLock.unlock()
+        }
+    }
+
+    /**
+     * @param lockProcessing       - returns what to do with Lock in collection
+     * @param resultPostProcessing - does some work after processing lock and applied action
+     */
+    fun processAllLocks(
+            lockProcessing: (LockIdentity, PersistentExpiringDistributedLock) -> ProcessingLockResult,
```

Comment:
You declared `processAllLocks(lockProcessing)` in a way that say to the user: define your `lockProcessing` function that should return KEEP_LOCK_IN_CONTAINER, REMOVE_LOCK_FROM_CONTAINER or ALREADY_REMOVED. 

But user actually should define `lockProcessing` function  that can return only KEEP_LOCK_IN_CONTAINER or REMOVE_LOCK_FROM_CONTAINER. 
Why you pushed `ALREADY_REMOVED` to the user api enum? What is the meaning behind that case?

Reply:
Agree. It's unnecessary

fixed